### DISCUSSION TITLE
Strange return values for npm outdated

### DISCUSSION THREAD

**Discussion by knev** [npm/feedback#931] [CLI]
`npm v9.6.6`

I'm currently scripting `npm` and I noticed the return value for `npm outdated` is odd.
When no package is installed, outdated prints nothing with zero as the return value. Which could be right, if outdated printed nothing and returned a positive number on failure. 

If a package is installed that needs to be upgraded, outdated prints and then the return value is ... 1, not zero.

**Comment by ljharb** [comment]
The success condition is "nothing needs updating", so that's when you get a zero. When something needs updating, you get nonzero. Are you seeing something different?

**Reply by knev** [reply]
I see. That's logical. But, then `npm outdated` never returns an error during execution?

**Reply by ljharb** [reply]
"returns an error" would mean "exits nonzero" - i'm not sure what you're asking?

**Reply by knev** [reply]
If `npm outdated` exists nonzero on error and also nonzero when a package needs updating, then the way to differentiate between those two, is if output was generated or not. 

This might be worth adding to the docs.
